src/andromede/expression/indexing.py

```diff
@@ -76,10 +75,11 @@ def negation(self, node: NegationNode) -> IndexingStructure:
         return visit(node.operand, self)
 
     def addition(self, node: AdditionNode) -> IndexingStructure:
-        return visit(node.left, self) | visit(node.right, self)
-
-    def substraction(self, node: SubstractionNode) -> IndexingStructure:
-        return visit(node.left, self) | visit(node.right, self)
+        operands = [visit(o, self) for o in node.operands]
```

Comment:
Dans 99% des cas, on va avoir des expressions indexées par le temps et les scénarios, on pourrait s'amuser ici à ne pas visiter tous les sous-arbres mais s'arrêter dès qu'un d'entre eux est effectivement indexé par (t,w). Le gain de perf est probablement négligeable mais comme parfois on a des surprises, je note la possible amélioration ici

Reply:
Peut-être que sur des expressions créées avec un nombre important de connexions à un même port, on évite un certain nombre de parcours

Reply:
Indeed I re-implemented it to stop as soon as we have a node which is both time and scenario dependent.